internal/controller/compute/statefulserverset/statefulserverset.go

```diff
@@ -109,55 +108,90 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
-		ResourceExists: creationLansUpToDate && creationVolumesUpToDate,
+		ResourceExists: areResourcesCreated,
 
 		// Return false when the externalStatefulServerSet resource exists, but it not up to date
 		// with the desired managed resource state. This lets the managed
 		// resource reconciler know that it needs to call Update.
-		ResourceUpToDate: areLansUpToDate && areVolumesUpToDate && isSSetUpToDate,
+		ResourceUpToDate: areResourcesUpdated,
 
 		// Return any details that may be required to connect to the externalStatefulServerSet
 		// resource. These will be stored as the connection secret.
 		ConnectionDetails: managed.ConnectionDetails{},
 	}, nil
 }
 
-func setSSetStatusOnCR(cr *v1alpha1.StatefulServerSet, sSet *v1alpha1.ServerSet) {
-	cr.Status.AtProvider.ReplicaStatus = sSet.Status.AtProvider.ReplicaStatuses
-	cr.Status.AtProvider.Replicas = sSet.Status.AtProvider.Replicas
+func (e *external) observeResourcesUpdateStatus(ctx context.Context, cr *v1alpha1.StatefulServerSet) (areResourcesCreated, areResourcesUpdated bool, err error) {
+
+	// ******************* LANS *******************
+	lans, err := e.LANController.ListLans(ctx, cr)
+	if err != nil {
+		return false, false, fmt.Errorf("while listing lans %w", err)
+	}
+	creationLansUpToDate, areLansUpToDate := areLansUpToDate(cr, lans.Items)
+	cr.Status.AtProvider.LanStatuses = setLanStatuses(lans.Items)
+
+	// ******************* VOLUMES *******************
+	volumes, err := e.dataVolumeController.ListVolumes(ctx, cr)
+	if err != nil {
+		return false, false, fmt.Errorf("while listing volumes %w", err)
+	}
+	creationVolumesUpToDate, areVolumesUpToDate := areDataVolumesUpToDate(cr, volumes.Items)
+	cr.Status.AtProvider.DataVolumeStatuses = setVolumeStatuses(volumes.Items)
+
+	// ******************* SERVERSET *******************
+	creationServerSetUpToDate, isServerSetUpToDate, err := e.isServerSetUpToDate(ctx, cr)
+	if err != nil {
+		return false, false, err
+	}
+	err = e.setSSetStatusOnCR(ctx, cr)
+	if err != nil {
+		return false, false, err
+	}
+
+	e.log.Info("Observing the StatefulServerSet", "creationLansUpToDate", creationLansUpToDate, "areLansUpToDate", areLansUpToDate, "creationVolumesUpToDate", creationVolumesUpToDate,
+		"areVolumesUpToDate", areVolumesUpToDate, "creationServerSetUpToDate", creationServerSetUpToDate, "isServerSetUpToDate", isServerSetUpToDate)
+
+	return creationLansUpToDate && creationVolumesUpToDate && creationServerSetUpToDate, areLansUpToDate && areVolumesUpToDate && isServerSetUpToDate, nil
+}
+
+func (e *external) isServerSetUpToDate(ctx context.Context, cr *v1alpha1.StatefulServerSet) (creationServerUpToDate bool, serverUpToDate bool, err error) {
+	_, err = e.SSetController.Get(ctx, getSSetName(cr), cr.Namespace)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			return false, false, nil
+		}
+	}
+	serverUpToDate, err = areSSetResourcesUpToDate(ctx, e.kube, cr)
+	if err != nil {
+		return true, true, err
```

Comment:
is `true` here a typo?

Reply:
it basically doesn't matter as the error will prevent anything from happening